transport/go.mod

```
@@ -0,0 +1,3 @@
+module github.com/fluxcd/pkg/transport
+
+go 1.18
```

We may want to keep the same golang version across all submodules, and upgrade them at the same time.

```suggestion
go 1.17
```